### PR TITLE
(Archiving) Add archive location tag to spring files

### PR DESCRIPTION
**Pull request by islean**
## Description

Part of issue #2716 where it was determined that Spring files should be tagged with the customer's archive location, so that the new Archiving/DDN/Miria flow knows whether to handle the files or not.

### Changed

- Spring files are now tagged with the customer's archive location in Housekeeper.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b archiving-add-archive-location-tag -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by henrikstranneheim — APPROVED** [review]
👍

File: cg/cli/compress/fastq.py [hunk 1]
```diff
@@ -92,7 +92,10 @@ def clean_fastq(context: CGConfig, case_id: str | None, days_back: int, dry_run:
     for case in cases:
         samples: Iterable[str] = store.get_sample_ids_by_case_id(case_id=case.internal_id)
         for sample_id in samples:
```
Comment (diitaz93) [inline cg/cli/compress/fastq.py:94]:
```suggestion
        sample_ids: Iterable[str] = store.get_sample_ids_by_case_id(case_id=case.internal_id)
        for sample_id in sample_ids:
```

File: cg/cli/add.py [hunk 2]
```diff
@@ -58,7 +59,7 @@ def add():
     "--data-archive-location",
     "data_archive_location",
     help="Specifies where to store data for the customer.",
-    default="PDC",
```
Comment (diitaz93) [inline cg/cli/add.py:61]: Im wondering if there should be a check that these locations are within a list of valid values. What if by accident you give "fastq" as a location? there will be spring files tagged as fast

**Review by diitaz93 — APPROVED** [review]
Neat 💯

**Comment by islean** [comment]
Testing went well:

image

And the spring file was tagged (ignore the second spring file):

image

File: cg/cli/add.py (hunk 2 above)
Comment (islean) [reply cg/cli/add.py:61]: This is not part of this flow, I just added the constant and exchanged the hard coded string. So this does not affect the tagging, this is for when you add a customer. I don't see a way to add a restriction to that @diitaz93

**Pull request merged**